packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -316,22 +516,330 @@ describe('Dropdown', () => {
+      wrapper
+        .find(`button.${Dropdown.slotClassNames.triggerButton}`)
+        .simulate('keydown', { keyCode: keyboardKey.ArrowUp, key: 'ArrowUp' })
+
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          highlightedIndex: 0,
+          open: true,
+        }),
+      )
+    })
+
+    it('is changed correctly on arrow down navigation', () => {
+      const wrapper = mountWithProvider(<Dropdown items={items} />)
+      const dropdown = wrapper.find(Dropdown)
+      const itemsList = wrapper.find(`ul.${Dropdown.slotClassNames.itemsList}`)
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+      for (let index = 0; index < 2; index++) {
+        itemsList.simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
+
+        expect(dropdown.state('highlightedIndex')).toBe(index)
+      }
+    })
+
+    it('is changed correctly on arrow up navigation', () => {
+      const wrapper = mountWithProvider(<Dropdown items={items} />)
+      const dropdown = wrapper.find(Dropdown)
+      const itemsList = wrapper.find(`ul.${Dropdown.slotClassNames.itemsList}`)
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+      for (let index = 0; index < 2; index++) {
+        itemsList.simulate('keydown', { keyCode: keyboardKey.ArrowUp, key: 'ArrowUp' })
+
+        expect(dropdown.state('highlightedIndex')).toBe(items.length - 1 - index)
+      }
+    })
+
+    it('wraps to start and end on navigation', () => {
+      const wrapper = mountWithProvider(<Dropdown items={items} defaultHighlightedIndex={1} />)
+      const dropdown = wrapper.find(Dropdown)
+      const itemsList = wrapper.find(`ul.${Dropdown.slotClassNames.itemsList}`)
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+      itemsList
+        .simulate('keydown', { keyCode: keyboardKey.ArrowUp, key: 'ArrowUp' })
+        .simulate('keydown', { keyCode: keyboardKey.ArrowUp, key: 'ArrowUp' })
+
+      expect(dropdown.state('highlightedIndex')).toBe(items.length - 1)
+
+      itemsList.simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
+
+      expect(dropdown.state('highlightedIndex')).toBe(0)
+    })
+  })
+
+  describe('value', () => {
+    it('is set by clicking on item', () => {
+      const itemSelectedIndex = 2
+      const onSelectedChange = jest.fn()
+      const wrapper = mountWithProvider(
+        <Dropdown onSelectedChange={onSelectedChange} items={items} />,
+      )
+
+      wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`).simulate('click')
+      wrapper
+        .find(`li.${Dropdown.slotClassNames.item}`)
+        .at(itemSelectedIndex)
+        .simulate('click')
+
+      expect(onSelectedChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          value: items[itemSelectedIndex],
```

Comment:
```suggestion
          value: 'item2',
```

Nit: It's more human readable